### PR TITLE
api: swagger: use explicit definitions, use inline examples, and fix indentation.

### PR DESCRIPTION
**Pull request by thaJeztah**
Some commits from https://github.com/moby/moby/pull/43332, leaving out the List response (which was problematic), but including a commit to carry https://github.com/moby/moby/pull/41748

- carries / closes https://github.com/moby/moby/pull/41748
- addresses https://github.com/moby/moby/issues/42593

Generating the `ContainerWaitResponse` as explicit types, makes them land in separate files, therefore "fixes" the CI flakiness for that type where the order of the types within the same file would randomise.

### CODE REVIEW

File: api/swagger.yaml [hunk 1]
```diff
@@ -9053,38 +9091,7 @@ paths:
...
-                type: "object"
-                additionalProperties:
-                  type: "string"
-            example:
-              Name: "tardis"
-              Labels:
-                com.example.some-label: "some-value"
-                com.example.some-other-label: "some-other-value"
-              Driver: "custom"
+            $ref: "#/definitions/VolumeCreateOptions"
```
Comment (tianon) [inline api/swagger.yaml:9103]: I feel like I'm missing something "obvious" about Swagger here, so please feel free to point me to documentation or something, but it seems strange to extract this out when there's only one use, especially since it also comes with code changes? :see_no_evil:

File: api/swagger.yaml [hunk 2]
```diff
@@ -7192,22 +7215,7 @@ paths:
...
-                type: "integer"
-                x-nullable: false
-              Error:
-                description: "container waiting error, if any"
-                type: "object"
-                properties:
-                  Message:
-                    description: "Details of an error"
-                    type: "string"
+            $ref: "#/definitions/ContainerWaitResponse"
```
Comment (tianon) [inline api/swagger.yaml:7219]: Ahhhhh, this commit mentions the CI issue -- is there a way we could somehow get the Swagger reproducibility fix backported to the version we use instead?

**Review by tianon — APPROVED** [review]
Generally seems OK, but I don't love that moving to definitions requires code changes / restructuring to be fully useful? :see_no_evil:

File: api/swagger.yaml (hunk 2 above)
Comment (thaJeztah) [reply api/swagger.yaml:7219]: The tricky bit is that effectively "we're using it wrong"; these types are generated for endpoints, and the swagger generator is meant to produce a full client for that (client, code handling the endpoint, as well as validation and models), and we are only using it to generate some of the types / structs.

To be honest, I think we should consider to do the _reverse_; generate the swagger from our Go structs (as we're generally better at defining neat Go structs than producing YAML), also, we're constantly trying to adjust the Swagger to make it "do the right thing" (and produce Go structs the way we want them to be).

Comment (thaJeztah) [reply api/swagger.yaml:7219]: The swagger "generated code" is also a bit awkward, in that it dumps everything in a single "types" package, which is an ill fit for Go conventions, and also, well, makes it a large bunch of (sometimes very unrelated) structs.

That said; moving the definition to the "definitions" inside the swagger may be slightly neater than "where did this weirdly named struct come from?", and (possibly) would allow some of those structs to _actually_ be reused (although, again, we may need to juggle things to make swagger do what we want 😞)

In short; I like swagger for it to generate nice looking docs, but I hate the generated bits (it's pretty poorly defined Go).

**Review by ndeloof — APPROVED** [review]

**Comment by thaJeztah** [comment]
Thx for review! Let me get this one in
